addons/mrp/models/stock_move.py

```diff
@@ -127,23 +127,10 @@ def _compute_order_finished_lot_ids(self):
         for move in self:
             move.order_finished_lot_ids = move.raw_material_production_id.lot_producing_id
 
-    @api.depends('raw_material_production_id.bom_id')
+    @api.depends('raw_material_production_id.bom_id.operation_ids')
     def _compute_allowed_operation_ids(self):
         for move in self:
-            if (
-                not move.raw_material_production_id or
-                not move.raw_material_production_id.bom_id or not
-                move.raw_material_production_id.bom_id.operation_ids
-            ):
-                move.allowed_operation_ids = self.env['mrp.routing.workcenter']
-            else:
-                operation_domain = [
-                    ('id', 'in', move.raw_material_production_id.bom_id.operation_ids.ids),
-                    '|',
-                        ('company_id', '=', self.company_id.id),
-                        ('company_id', '=', False)
-                ]
-                move.allowed_operation_ids = self.env['mrp.routing.workcenter'].search(operation_domain)
+            move.allowed_operation_ids = move.raw_material_production_id.bom_id.operation_ids.ids
```

Comment:
ça pose pas de soucis si raw_material_production_id n'est pas défini ?

Reply:
Not, I don't think so, the implicite mapped will do the job. But the ids at the end can be remove :+1: